src_feedback/elf.rs

```diff
@@ -0,0 +1,666 @@
+#[allow(dead_code)]
+pub const ET_DYN: u16 = 3;
+#[allow(dead_code)]
+pub const ET_CORE: u16 = 4;
+#[allow(dead_code)]
+pub const ET_LOPROC: u16 = 0xff00;
+#[allow(dead_code)]
+pub const ET_HIPROC: u16 = 0xffff;
+
+#[repr(C)]
+struct Elf64_Ehdr {
+    e_ident: [u8; EI_NIDENT],	/* ELF "magic number" */
+    e_type: Elf64_Half,
+    e_machine: Elf64_Half,
+    e_version: Elf64_Word,
+    e_entry: Elf64_Addr,	/* Entry point virtual address */
+    e_phoff: Elf64_Off,		/* Program header table file offset */
+    e_shoff: Elf64_Off,		/* Section header table file offset */
+    e_flags: Elf64_Word,
+    e_ehsize: Elf64_Half,
+    e_phentsize: Elf64_Half,
+    e_phnum: Elf64_Half,
+    e_shentsize: Elf64_Half,
+    e_shnum: Elf64_Half,
+    e_shstrndx: Elf64_Half,
+}
+
+#[allow(dead_code)]
+pub const PT_NULL: u32 = 0;
+#[allow(dead_code)]
+pub const PT_LOAD: u32 = 1;
+#[allow(dead_code)]
+pub const PT_DYNAMIC: u32 = 2;
+#[allow(dead_code)]
+pub const PT_INTERP: u32 = 3;
+#[allow(dead_code)]
+pub const PT_NOTE: u32 = 4;
+#[allow(dead_code)]
+pub const PT_SHLIB: u32 = 5;
+#[allow(dead_code)]
+pub const PT_PHDR: u32 = 6;
+#[allow(dead_code)]
+pub const PT_TLS: u32 = 7;               /* Thread local storage segment */
+#[allow(dead_code)]
+pub const PT_LOOS: u32 = 0x60000000;      /* OS-specific */
+#[allow(dead_code)]
+pub const PT_HIOS: u32 = 0x6fffffff;      /* OS-specific */
+#[allow(dead_code)]
+pub const PT_LOPROC: u32 = 0x70000000;
+#[allow(dead_code)]
+pub const PT_HIPROC: u32 = 0x7fffffff;
+#[allow(dead_code)]
+pub const PT_GNU_EH_FRAME: u32 = 0x6474e550;
+#[allow(dead_code)]
+pub const PT_GNU_PROPERTY: u32 = 0x6474e553;
+
+#[allow(dead_code)]
+pub const PT_GNU_STACK: u32 = PT_LOOS + 0x474e551;
+
+#[repr(C)]
+pub struct Elf64_Phdr {
+    pub p_type: Elf64_Word,
+    pub p_flags: Elf64_Word,
+    pub p_offset: Elf64_Off,	/* Segment file offset */
+    pub p_vaddr: Elf64_Addr,	/* Segment virtual address */
+    pub p_paddr: Elf64_Addr,	/* Segment physical address */
+    pub p_filesz: Elf64_Xword,	/* Segment size in file */
+    pub p_memsz: Elf64_Xword,	/* Segment size in memory */
+    pub p_align: Elf64_Xword,	/* Segment alignment, file & memory */
+}
+
+#[repr(C)]
+struct Elf64_Shdr {
+    sh_name: Elf64_Word,	/* Section name, index in string tbl */
+    sh_type: Elf64_Word,	/* Type of section */
+    sh_flags: Elf64_Xword,	/* Miscellaneous section attributes */
+    sh_addr: Elf64_Addr,	/* Section virtual addr at execution */
+    sh_offset: Elf64_Off,	/* Section file offset */
+    sh_size: Elf64_Xword,	/* Size of section in bytes */
+    sh_link: Elf64_Word,	/* Index of another section */
+    sh_info: Elf64_Word,	/* Additional section information */
+    sh_addralign: Elf64_Xword,	/* Section alignment */
+    sh_entsize: Elf64_Xword,	/* Entry size if section holds table */
+}
+
+pub const SHN_UNDEF: u16 = 0;
+
+#[allow(dead_code)]
+pub const STT_NOTYPE: u8 = 0;
+#[allow(dead_code)]
+pub const STT_OBJECT: u8 = 1;
+#[allow(dead_code)]
```

Comment:
I think we can remove this annotation since STT_FUNC is used

Also, do you think we need STT_NOTYPE and STT_OBJECT defined here or should these be removed since they are not used anywhere?

Reply:
Yes, we should revise these annotations.

These constants and types are unlikely to be changed.  They are part of the standard or de-facto.  Maybe we keep them in a separate file and forget it.